### PR TITLE
Add support for RFC 3114, RFC 5755, RFC 5913, and RFC 5917

### PR DESCRIPTION
**Pull request by russhousley**
Add modules and tests for RFC 3114, RFC 5755, RFC 5913, and RFC 5917.

### CODE REVIEW

**Comment by etingof** [comment]
CI fails on Py3 jobs:

```
======================================================================
FAIL: testOpenTypes (tests.test_rfc5755.AttributeCertificateTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/etingof/pyasn1-modules/tests/test_rfc5755.py", line 85, in testOpenTypes
    assert attr['values'][0]['authInfo'] == 'password'
AssertionError
```

May be this has something to do with `str` vs `bytes` in "password"?

**Comment by russhousley** [comment]
I am also seeing this when I run under python3:

.../pyasn1/type/univ.py:711: DeprecationWarning: __int__ returned non-int (type SizedInteger).  The ability to return an instance of a strict subclass of int is deprecated, and may be removed in a future version of Python.
  return SizedInteger(value).setBitLength(len(value))

**Comment by russhousley** [comment]
I added str() so that it works under python2 and python3.

**Comment by etingof** [comment]
> I added str() so that it works under python2 and python3.

I changed it to `str2octs` so that we would ensure the result is in `bytes` on Py3.

File: pyasn1_modules/rfc3114.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from pyasn1_modules import rfc5755
+
+
+id_smime = univ.ObjectIdentifier((1, 2, 840, 113549, 1, 9, 16, ))
+
+id_tsp = id_smime + (7, )
+
+id_tsp_TEST_Amoco = id_tsp + (1, )
+
+class Amoco_SecurityClassification(univ.Integer):
```
Comment (etingof) [inline pyasn1_modules/rfc3114.py]: nitpick: it so happened that we primarily use CamelCase (no underscore), so it probably makes sense to stick with it if possible.

**Comment by etingof** [comment]
Thank you!

File: pyasn1_modules/rfc3114.py (hunk 1 above)
Comment (russhousley) [reply pyasn1_modules/rfc3114.py]: This is to align with the ASN.1 definition with hyphen changed to underscore:

   Amoco-SecurityClassification ::= INTEGER {
     amoco-general (6),
     amoco-confidential (7),
     amoco-highly-confidential (8) }

Comment (etingof) [reply pyasn1_modules/rfc3114.py]: I assume that hyphen does not occur often in ASN.1 definitions. This is probably the first time when it caught my eye.

Comment (russhousley) [reply pyasn1_modules/rfc3114.py]: It happens all the time, especially with OIDs.  asn1ate does the substitution from hypen to underscore automatically.  The following examples are at the top of the ASN.1 module in RFC 5280...

-- PKIX specific OIDs

id-pkix  OBJECT IDENTIFIER  ::=
         { iso(1) identified-organization(3) dod(6) internet(1)
                    security(5) mechanisms(5) pkix(7) }

-- PKIX arcs

id-pe OBJECT IDENTIFIER ::= { id-pkix 1 }
        -- arc for private certificate extensions
id-qt OBJECT IDENTIFIER ::= { id-pkix 2 }
        -- arc for policy qualifier types
id-kp OBJECT IDENTIFIER ::= { id-pkix 3 }
        -- arc for extended key purpose OIDS
id-ad OBJECT IDENTIFIER ::= { id-pkix 48 }
        -- arc for access descriptors

-- policyQualifierIds for Internet policy qualifiers

id-qt-cps      OBJECT IDENTIFIER ::=  { id-qt 1 }
      -- OID for CPS qualifier
id-qt-unotice  OBJECT IDENTIFIER ::=  { id-qt 2 }
      -- OID for user notice qualifier

-- access descriptor definitions

id-ad-ocsp         OBJECT IDENTIFIER ::= { id-ad 1 }
id-ad-caIssuers    OBJECT IDENTIFIER ::= { id-ad 2 }
id-ad-timeStamping OBJECT IDENTIFIER ::= { id-ad 3 }
id-ad-caRepository OBJECT IDENTIFIER ::= { id-ad 5 }

**Pull request merged**